packages/api/amplify_api/lib/src/decorators/authorize_http_request.dart

```diff
@@ -64,8 +64,15 @@ Future<http.BaseRequest> authorizeHttpRequest(http.BaseRequest request,
       return authorizedRequest.httpRequest;
     case APIAuthorizationType.function:
     case APIAuthorizationType.oidc:
-    case APIAuthorizationType.userPools:
       throw UnimplementedError('${authType.name} not implemented.');
```

Comment:
I haven't tested with OIDC/lambda, but I'm thinking when we register the API auth providers from API plugin instantiation with the auth provider repo which define different way to get token, we can just remove this line here and this part should work with those providers.